runtime/context.go

```diff
@@ -99,6 +100,25 @@ func AnnotateIncomingContext(ctx context.Context, mux *ServeMux, req *http.Reque
 	return metadata.NewIncomingContext(ctx, md), nil
 }
 
+func isValidGRPCMetadataKey(key string) bool {
+	for i := 0; i < len(key); i++ {
+		r := key[i]
+		if !(r >= 'a' && r <= 'z') && !(r >= '0' && r <= '9') && r != '.' && r != '-' && r != '_' {
```

Comment:
Please reference https://github.com/grpc/grpc/blob/4b05dc88b724214d0c725c8e7442cbc7a61b1374/doc/PROTOCOL-HTTP2.md in a comment here to make it clear where this requirement is from. The exact definition I see is

```
%x30-39 / %x61-7A / "_" / "-" / "."
```
or
```
0-9 a-z _ - .
```

Reply:
Done.